txscript/opcode.go

```diff
@@ -1850,7 +1850,17 @@ func opcodeCheckSig(op *parsedOpcode, vm *Engine) error {
 		return nil
 	}
 
-	ok := signature.Verify(hash, pubKey)
+	var ok bool
+	if vm.sigCache != nil {
+		ok = vm.sigCache.Exists(hash, signature, pubKey)
```

Comment:
For this same dance below, in opcodeCheckMultiSig, you use valid instead of ok.  Better to keep them consistent, better yet to push this out to a new function.